### PR TITLE
Enhancement: History (audit log) for bulk edit operations

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Closes #7569

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by nedy13** [comment]
I have no idea about the rest of the code, but I just noticed that there is no test included if the feature is off, that it is off.

Case `@override_settings(AUDIT_LOG_ENABLED=False)` and no LogEntry is added.

**Comment by shamoon** [comment]
Not sure there’s really to test that, since the only functional modifications in the flow here are wrapped in a very simple “if enabled”…

**Comment by stumpylog** [comment]
We're not trying to achieve branch coverage

File: src/documents/views.py [hunk 1]
```diff
@@ -1013,8 +1031,54 @@ def post(self, request, *args, **kwargs):
...
+                    ),
+                )
+
             # TODO: parameter validation
             result = method(documents, **parameters)
+
+            if settings.AUDIT_LOG_ENABLED and modified_field:
+                new_documents = Document.objects.filter(pk__in=documents)
+                for doc in new_documents:
+                    old_value = next(
```
Comment (stumpylog) [inline src/documents/views.py:1055]: I'm not sure how much we're concerned about performance, but I think this has to go through the list each loop iteration.

Maybe a dict like `old_documents = {obj.pk: obj for obj in queryset}`?

Comment (shamoon) [reply src/documents/views.py]: Performance always a good goal, thanks!

**Review by stumpylog — APPROVED** [review]

**Pull request merged**